service.go

```diff
@@ -238,105 +361,171 @@ func (client *Client) GetServiceCount() (int, error) {
 	return int(q.Account.Services.TotalCount), HandleErrors(err, nil)
 }
 
-// TODO: maybe we can find a way to merge ServiceConnection.Query & Hydrate
-func (conn *ServiceConnection) Query(client *Client, q interface{}, v PayloadVariables) ([]Service, error) {
-	if err := client.Query(q, v, WithName("ServiceList")); err != nil {
-		return conn.Nodes, err
-	}
-	if err := conn.Hydrate(client); err != nil {
-		return conn.Nodes, err
-	}
-	return conn.Nodes, nil
-}
-
-func (conn *ServiceConnection) Hydrate(client *Client) error {
+func (client *Client) ListServices(variables *PayloadVariables) (ServiceConnection, error) {
 	var q struct {
 		Account struct {
 			Services ServiceConnection `graphql:"services(after: $after, first: $first)"`
 		}
 	}
-	v := PayloadVariables{
-		"first": client.pageSize,
+	if variables == nil {
+		variables = client.InitialPageVariablesPointer()
 	}
-	q.Account.Services.PageInfo = conn.PageInfo
+
+	if err := client.Query(&q, *variables, WithName("ServiceList")); err != nil {
+		return ServiceConnection{}, err
+	}
+
 	for q.Account.Services.PageInfo.HasNextPage {
-		v["after"] = q.Account.Services.PageInfo.End
-		if err := client.Query(&q, v, WithName("ServiceList")); err != nil {
-			return err
-		}
-		for _, item := range q.Account.Services.Nodes {
-			if err := (&item).Hydrate(client); err != nil {
-				return err
-			}
-			conn.Nodes = append(conn.Nodes, item)
+		(*variables)["after"] = q.Account.Services.PageInfo.End
+		resp, err := client.ListServices(variables)
+		if err != nil {
+			return ServiceConnection{}, err
 		}
+		q.Account.Services.Nodes = append(q.Account.Services.Nodes, resp.Nodes...)
```

Comment:
I think we need to loop through each service - call hydrate on it and then stuff it into the array